Sources/Purchasing/Purchases/Purchases.swift

```diff
@@ -2322,24 +2322,28 @@ private extension Purchases {
             isAppBackgrounded: isAppBackgrounded
         ) { [weak self] offeringsResultData in
             if #available(iOS 15.0, macOS 12.0, watchOS 8.0, tvOS 15.0, *),
-               let self = self, let cache = self.paywallCache, let offerings = offeringsResultData.value?.offerings {
-                self.operationDispatcher.dispatchOnWorkerThread {
-                    await withTaskGroup(of: Void.self) { group in
-                        group.addTask {
-                            await cache.warmUpEligibilityCache(offerings: offerings)
-                        }
-                        group.addTask {
-                            await cache.warmUpPaywallImagesCache(offerings: offerings)
-                        }
-						group.addTask {
-                            await cache.warmUpPaywallFontsCache(offerings: offerings)
-                        }
-                    }
-                }
+               let offerings = offeringsResultData.value?.offerings {
+                self?.warmUpCaches(offerings: offerings)
             }
         }
     }
 
+    @available(iOS 15.0, macOS 12.0, watchOS 8.0, tvOS 15.0, *)
+    private func warmUpCaches(offerings: Offerings) {
+        guard let cache = self.paywallCache else {
+            return
+        }
+        self.operationDispatcher.dispatchOnWorkerThread {
+            await cache.warmUpEligibilityCache(offerings: offerings)
+        }
+        self.operationDispatcher.dispatchOnWorkerThread {
+            await cache.warmUpPaywallImagesCache(offerings: offerings)
+        }
+        self.operationDispatcher.dispatchOnWorkerThread {
+            await cache.warmUpPaywallFontsCache(offerings: offerings)
+        }
+    }
```

Comment:
Looking at the implementations of these three (and the crash), I wonder if we should guard against `Task.isCancelled` to be on the safe side.

Reply:
Yep. That makes total sense. And something I also thought about, but decided to simplify things and make minimal changes to the behavior.

In fact, in order for that to work, we would need to change the calls to `dispatchOnWorkerThread`, because they create a `Task.detached`, which means that cancellation is not inherited. https://github.com/RevenueCat/purchases-ios/blob/12ae80ac0fb7b5be2dff93a8d353a8174792ff6b/Sources/Misc/Concurrency/OperationDispatcher.swift#L68-L77

In addition, it made sense to me, in this case, to have the tasks be detached, because we're only warming up caches, which makes sense to do even if the parent task gets cancelled I'd say. I don't have a strong opinion though, so I'm open to changing this 👍